src/components/Attachments/AttachmentCarousel/index.js

```diff
@@ -38,6 +40,19 @@ function AttachmentCarousel({report, reportActions, source, onNavigate, setDownl
     const [activeSource, setActiveSource] = useState(source);
     const [shouldShowArrows, setShouldShowArrows, autoHideArrows, cancelAutoHideArrows] = useCarouselArrows();
 
+    const isLoadingReport = isLoadingReportData && (_.isEmpty(report) || !report.reportID);
+    const isLoadingReportAction = _.isEmpty(reportActions) && reportMetadata.isLoadingReportActions;
+
+    // For small screen, we don't call openReport API when we go to a sub report page by deeplink
+    // So we need to call openReport here for small screen
+    useEffect(() => {
+        if (!isSmallScreenWidth || (!_.isEmpty(report) && !_.isEmpty(reportActions))) {
+            return;
+        }
+        Report.openReport(report.reportID);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isSmallScreenWidth, report.reportID]);
+
```

Comment:
We usually avoid duplicated code. What do you think about adding this logic in parent component - AttachmentModal?